Source/common/new.santa.proto

```diff
@@ -0,0 +1,283 @@
+syntax = "proto3";
+
+import "google/protobuf/any.proto";
+
+option objc_class_prefix = "SNTPB";
+
+package santa;
+
+message UserInfo {
+  int32 uid = 1;
+  string name = 2;
+}
+
+message GroupInfo {
+  int32 gid = 1;
+  string name = 2;
+}
+
+message ProcessID {
+  int32 pid = 1;
+  int32 pidversion = 2;
+}
+
+message NanoTime {
+    int64 sec;
+    int64 nsec;
+}
+
+message MicroTime {
+    int64 sec;
+    int64 usec;
+}
+
+message CodeSignature {
+  bytes cdhash = 1;
+  string signing_id = 2;
+  string team_id = 3;
+}
+
+message Stat {
+  int32 dev = 1;
+  uint32 mode = 2;
+  uint32 nlink = 3;
+  uint64 ino = 4;
+  UserInfo user = 5;
+  GroupInfo group = 6;
+  int32 rdev = 7;
+  NanoTime a_time = 8;
+  NanoTime m_time = 9;
+  NanoTime c_time = 10;
+  NanoTime b_time = 11;
```

Comment:
Expand out the field names. 

Also, are all 4 of these useful to record? I didn't think macOS ever recorded access times and I'm unsure whether there's value in recording both modify and change (the latter seems more useful, to me). Each one we record is up to 17 bytes, which adds up with the number of events we generate.

Reply:
It can be useful, really depends on the context and the type of signal someone is looking for.

This proto is definitely more verbose that what Santa has historically logged. The assumption is that when deployed, it's on system with reasonable upload bandwidth and proper log rotation in place. Personally, I'd hesitate to try and eliminate specific fields from things like `stat` to save a few bytes  given it could be deployed in an environment where that signal was important. But I understand the argument and happy to chat more offline.

Reply:
Fair enough, then just expand the field names.